doc/userguide/devguide/contributing/code-submission-process.rst

```diff
@@ -9,16 +9,20 @@ Commits
 #. Commits need to be logically separated. Don't fix unrelated things in one commit.
 #. Don't add unnecessary commits, if commit 2 fixes commit 1 merge them together (squash)
 #. Commits need to have proper messages, explaining anything that is non-trivial
-#. Commits should not at the same time change, rename and/or move code. Use separate commits
-   for each of this, e.g, a commit to rename files, then a commit to change the code.
-#. Documentation updates should be in their own commit (not mixed with code commits)
-#. Commit messages need to be properly formatted:
-    * Meaningful and short (50 chars max) subject line followed by an empty line
-    * Naming convention: prefix message with sub-system ("rule parsing: fixing foobar"). If
-      you're not sure what to use, look at past commits to the file(s) in your PR.
-    * Description, wrapped at ~72 characters
+#. Commits should not, at the same time, change, rename and/or move code. Use
+   separate commits for each of this, e.g, a commit to rename files, then a commit
+   to change the code.
+#. If your code changes or adds new behavior, add the related documentation
+   updates in their own commit, but make sure to add the same ticket number to
+   both commit messages.
```

Comment:
Have we pivoted on this? Or do we still want to stick with it.

Reply:
Victor says he doesn't have a strong opinion on this, but feels it's better to have smaller commits, due to backporting processes. So I figured having the ticket reference would be a middle ground for it.
He said that if anyone felt strongly about it, we could change. I'll follow the lead, I just want to have something easy for folks to understand and follow, and for us to have less chances of missing stuff...

Reply:
Ok, I think the practice is OK as long as a ticket links them. But sometimes we might have to look at it on a case by case basis as well.

Reply:
I have similar feelings. That's why I thought of having a checkpoint to sort of ensure no commit was left behind among several reviews of a given work.